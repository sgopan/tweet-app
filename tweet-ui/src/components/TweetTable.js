import React, { Component } from 'react';
import moment from 'moment';

class TweetTable extends Component {
    constructor(props) {
        super(props);
        this.state = {
          tweets: []
        }
      }
    
      componentDidMount() {
            fetch('http://localhost:3002/api/tweets')
        .then(res => res.json())
        .then(tweets => this.setState({ 'tweets': tweets }));
      }
      
      render() {
        const {tweets} = this.state;
        return ( 
            <div style={{ margin:"20px" }}>
            <h1 class="text-center">Tweet Table</h1>
            <table className="table table-striped" style={{ border:"1px solid #ddd" }}>
      <thead  style={{ backgroundColor:"#ADD8E6" }}>
        <tr>
          <th>Tweet Id</th>
          <th>Tweet</th>
          <th>User Id</th>
          <th>User Name</th>
          <th>Tweeted At</th>
        </tr>
      </thead>
      <tbody>
         { (tweets.length > 0) ? tweets.map( (tweet, index) => {
           return (
            <tr key={ index }>
              <td>{ tweet.tweet_id }</td>
              <td>{ tweet.tweet_message }</td>
              <td>{ tweet.user_id}</td>
              <td>{ tweet.user_name }</td>
              <td>{moment(tweet.createdAt).format('MMMM Do YYYY')}</td>
            </tr>
          )
         }) : <tr><td colSpan="5">Loading...</td></tr> }
      </tbody>
    </table>   
    </div>   
        );
      }
}


export default TweetTable