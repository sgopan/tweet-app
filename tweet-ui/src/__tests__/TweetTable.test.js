import React from 'react';
import TweetTable from '../components/TweetTable.js';
import { create } from 'react-test-renderer'

describe('Basic snapshot test',()=>{
    test('testing tweet table', () => {
    let tree = create(<TweetTable />)
    expect(tree.toJSON()).toMatchSnapshot();
  })
})