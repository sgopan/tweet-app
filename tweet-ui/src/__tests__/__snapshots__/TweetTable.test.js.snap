// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic snapshot test testing tweet table 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      Tweet Demo
    </a>
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="#home"
        href="#home"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Home
      </a>
    </div>
  </nav>
  <div
    className="table-responsive-lg"
  >
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Tweet Id
          </th>
          <th>
            Tweet
          </th>
          <th>
            User Id
          </th>
          <th>
            User Name
          </th>
          <th>
            Tweeted At
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            colSpan="5"
          >
            Loading...
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
